fvm/src/machine/engine.rs

```diff
@@ -105,13 +117,41 @@ pub fn default_wasmtime_config() -> wasmtime::Config {
     // > not enabled by default.
     c.cranelift_nan_canonicalization(true);
 
+    // wasmtime default: 512KiB
     // Set to something much higher than the instrumented limiter.
-    c.max_wasm_stack(64 << 20).unwrap();
+    // Note: This is in bytes, while the instrumented limit is in stack elements
+    c.max_wasm_stack(4 << 20).unwrap();
 
     // Execution cost accouting is done through wasm instrumentation,
     c.consume_fuel(false);
-
-    // c.cranelift_opt_level(Speed); ?
+    c.epoch_interruption(false);
+
+    // Disable debug-related things, wasm-instrument doesn't fix debug info
+    // yet, so those aren't useful, just add overhead
+    c.debug_info(false);
+    c.generate_address_map(false);
+    c.cranelift_debug_verifier(false);
+
+    // Reiterate some defaults
+    c.guard_before_linear_memory(true);
+    c.interruptable(false);
+    c.parallel_compilation(true);
+
+    c.allocation_strategy(InstanceAllocationStrategy::Pooling {
+        strategy: PoolingAllocationStrategy::NextAvailable,
+        instance_limits: InstanceLimits {
+            count: 1000, // todo set this higher than recursive call limit?
+            size: 1 << 20,
```

Comment:
In combination with the count, as far as I can see this will lead to pre-allocation of 1GiB of memory.
Are we fine with it?

https://docs.rs/wasmtime/latest/wasmtime/struct.InstanceLimits.html#structfield.size